### PR TITLE
Start adding a demo for opencensus-contrib-log-correlation-log4j.

### PR DESCRIPTION
**Pull request by sebright**
This PR is an early version of a demo for the library in #1371.  I'm planning to add the demo to https://github.com/census-ecosystem/opencensus-experiments after the library is released instead of merging this PR.  I also need to improve the readme.  ~~The first commit is from #1371.~~

### CODE REVIEW

File: contrib/log_correlation/log4j_demo/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Configuration status="WARN">
+  <Appenders>
+    <File name="File" fileName="/tmp/test-structured-log.log">
+      <JsonLayout compact="true" eventEol="true">
+
+        <!-- This section reads the trace ID, span ID, and sampled fields from the Log4j context -->
+        <!-- and add them to the JSON log entry as key-value pairs. It uses the special JSON -->
```
Comment (g-easy) [inline contrib/log_correlation/log4j_demo/src/main/resources/log4j2.xml]: add -> adds

Comment (g-easy) [inline contrib/log_correlation/log4j_demo/src/main/resources/log4j2.xml]: Can't have multi-line comments?

**Comment by sebright** [comment]
Thanks!  I addressed the two comments, and I'll clean this up further when #1371 is done.

**Comment by sebright** [comment]
I made a snapshot release of opencensus-java and moved this demo to opencensus-experiments: https://github.com/census-ecosystem/opencensus-experiments/pull/60

**Pull request closed without merging**